presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -55,103 +71,198 @@ public CatalogServer(MetadataManager metadataProvider, SessionPropertyManager se
+        this.getMaterializedViewCache = newCacheBuilder(cacheExpiresAfterWriteMillis, cacheMaximumSize).build(CacheLoader.from(this::loadGetMaterializedView));
+        this.getReferencedMaterializedViewsCache = newCacheBuilder(cacheExpiresAfterWriteMillis, cacheMaximumSize).build(CacheLoader.from(this::loadGetReferencedMaterializedViews));
+    }
+
+    /*
+        Loading Cache Methods
+     */
+
+    private Boolean loadCatalogExists(CacheKey key)
+    {
+        return metadataProvider.catalogExists(key.getSession().toSession(sessionPropertyManager), (String) key.getKey());
+    }
+
+    private Boolean loadSchemaExists(CacheKey key)
+    {
+        return metadataProvider.schemaExists(key.getSession().toSession(sessionPropertyManager), (CatalogSchemaName) key.getKey());
+    }
+
+    private List<String> loadListSchemaNames(CacheKey key)
+    {
+        return metadataProvider.listSchemaNames(key.getSession().toSession(sessionPropertyManager), (String) key.getKey());
     }
 
+    private Optional<TableHandle> loadGetTableHandle(CacheKey key)
+    {
+        return metadataProvider.getTableHandle(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    private List<QualifiedObjectName> loadListTables(CacheKey key)
+    {
+        return metadataProvider.listTables(key.getSession().toSession(sessionPropertyManager), (QualifiedTablePrefix) key.getKey());
+    }
+
+    private List<QualifiedObjectName> loadListViews(CacheKey key)
+    {
+        return metadataProvider.listViews(key.getSession().toSession(sessionPropertyManager), (QualifiedTablePrefix) key.getKey());
+    }
+
+    private Map<QualifiedObjectName, ViewDefinition> loadGetViews(CacheKey key)
+    {
+        return metadataProvider.getViews(key.getSession().toSession(sessionPropertyManager), (QualifiedTablePrefix) key.getKey());
+    }
+
+    private Optional<ViewDefinition> loadGetView(CacheKey key)
+    {
+        return metadataProvider.getView(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    private Optional<ConnectorMaterializedViewDefinition> loadGetMaterializedView(CacheKey key)
+    {
+        return metadataProvider.getMaterializedView(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    private List<QualifiedObjectName> loadGetReferencedMaterializedViews(CacheKey key)
+    {
+        return metadataProvider.getReferencedMaterializedViews(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    /*
+        Metadata Manager Methods
+     */
+
     @ThriftMethod
-    public boolean schemaExists(TransactionInfo transactionInfo, SessionRepresentation session, CatalogSchemaName schema)
+    public CatalogServerClient.MetadataEntry<Boolean> schemaExists(TransactionInfo transactionInfo, SessionRepresentation session, CatalogSchemaName schema)
```

Comment:
We should make a separate class for `MetadataEntry`, otherwise we need to at least `import static` it.

Reply:
I will opt for the separate class option as its used by several classes and can be generalized kind of, done